src/compas_timber/elements/fasteners/plate_fastener.py

```diff
@@ -0,0 +1,340 @@
+import math
+
+from compas.geometry import Brep
+from compas.geometry import Cylinder
+from compas.geometry import Frame
+from compas.geometry import NurbsCurve
+from compas.geometry import Plane
+from compas.geometry import Point
+from compas.geometry import Transformation
+from compas.geometry import Vector
+from compas.geometry import angle_vectors
+from compas.geometry import cross_vectors
+from compas.geometry import distance_point_plane
+from compas.tolerance import Tolerance
+
+from compas_timber.connections.utilities import beam_ref_side_incidence_with_vector
+from compas_timber.elements import Fastener
+from compas_timber.elements import FastenerApplicationError
+from compas_timber.elements import FastenerTimberInterface
+from compas_timber.utils import intersection_line_line_param
+
+TOL = Tolerance()
+
+
+class PlateFastener(Fastener):
+    """
+    A class to represent flat plate timber fasteners (e.g. steel plates).
+
+    Parameters
+    ----------
+    shape : :class:`~compas.geometry.Geometry`
+        The shape of the fastener at the XY plane origin.
+    frame : :class:`~compas.geometry.Frame`
+        The frame of the instance of the fastener that is applied to the model.
+        The fastener should be defined at the XY plane origin with the x-axis pointing in the direction of the main_beam.
+    holes : list of tuple, optional
+        The holes of the fastener. Structure is as follows: [(point, diameter), ...]
+    angle : float, optional (default=math.pi / 2)
+        The angle of the fastener. The angle between the beam elements must be the same.
+
+    Attributes
+    ----------
+    geometry : :class:`~compas.geometry.Geometry`
+        The geometry of the fastener.
+    frame : :class:`~compas.geometry.Frame`
+        The frame of the fastener.
+
+    """
+
+    def __init__(self, shape=None, frame=None, angle=math.pi / 2, topology=None, interfaces=[], outline = None, thickness = None, **kwargs):
```

Comment:
looking at the three ways to instantiate this class, make me feel like there room for revising how `PlateFastener` is defined, i.e. what are the basic attributes which constitute a `PlateFastener`.

specifically this `__init__`, the class method `from_outline_thickness_interfaces_cutouts` and the class method `default_T` all create an instance and while they have some overlaps they potentially create very different versions of instances of this class.

A good way to go about it is to define the core attributes that are essential to the creation of a `PlateFastener` and accept them in `__init__()`. it should always be possible to create a basic instance of it via the default init.

There might be some additional, optional, arguments that specify the fastener further, these should be accepted by `__init__` as well, as optional kwargs.

All of the above mentioned attributes should either already exist in parent classe(s) (e.g. `frame`) and therefore passed to `super` or they should be declared in the `__init__` body.

subsequently alternative constructors should provided some or all of these attributes (at least the core ones) to `__init__`, normally without creating new attributes.